base/ca/src/main/java/org/dogtagpki/server/ca/rest/base/ProfileBase.java

```diff
@@ -0,0 +1,1252 @@
+                    OpDef.OP_ADD,
+                    profileId,
+                    ILogger.SUCCESS,
+                    auditParams);
+
+            return profileId;
+
+        } catch (EBaseException | IOException e) {
+
+            String message = "Unable to create profile: " + e.getMessage();
+            logger.error(message, e);
+
+            auditProfileChange(
+                    ScopeDef.SC_PROFILE_RULES,
+                    OpDef.OP_ADD,
+                    profileId,
+                    ILogger.FAILURE,
+                    auditParams);
+
+            throw new PKIException(message, e);
+        }
+    }
+
+    public void modifyProfileState(Principal principal, String profileId, String action) throws EBaseException {
+        if (profileId == null) {
+            logger.error("modifyProfileState: invalid request. profileId is null");
+            throw new BadRequestException("Unable to modify profile state: Invalid Profile Id");
+        }
+
+        if (action == null) {
+            logger.error("modifyProfileState: invalid request. action is null");
+            throw new BadRequestException("Unable to modify profile state: Missing action");
+        }
+
+        if (ps == null) {
+            logger.error("modifyProfileState: ps is null");
+            throw new PKIException("Error modifying profile state.  Profile Service not available");
+        }
+
+        try {
+            Profile profile = ps.getProfile(profileId);
+            if (profile == null) {
+                logger.error("Trying to modify profile: {}.  Profile not found.", profileId);
+                throw new ProfileNotFoundException(profileId);
+            }
+        } catch (EProfileException e1) {
+            e1.printStackTrace();
+            throw new PKIException("Error modifying profile state: unable to get profile");
+        }
+
+        switch (action) {
+        case "enable":
+            if (ps.isProfileEnable(profileId)) {
+                throw new ConflictingOperationException("Profile already enabled");
+            }
+            try {
+                ps.enableProfile(profileId, principal.getName());
+                ps.commitProfile(profileId);
+                auditProfileChangeState(profileId, "approve", ILogger.SUCCESS);
+            } catch (EProfileException e) {
+                logger.error("modifyProfileState: error enabling profile: " + e.getMessage(), e);
+                auditProfileChangeState(profileId, "approve", ILogger.FAILURE);
+                throw new PKIException("Error enabling profile");
+            }
+            break;
+        case "disable":
+            if (!ps.isProfileEnable(profileId)) {
+                throw new ConflictingOperationException("Profile already disabled");
+            }
+            String userid = principal.getName();
+            try {
+                if (ps.checkOwner()) {
+                    if (ps.getProfileEnableBy(profileId).equals(userid)) {
```

Comment:
This is checking to see whether the profile was enabled by the same uid that's trying to disable it now.  I think the policy we want is to **check whether the person that's trying to disable the profile is a CA agent.**  Let's say if Joe was a CA agent who enabled a profile, but later his CA agent privilege was removed, then he can't be allowed to disable the profile.

Reply:
I believe this code is basically identical to the original code here:
https://github.com/dogtagpki/pki/blob/master/base/ca/src/main/java/org/dogtagpki/server/ca/rest/v1/ProfileService.java#L449-L475

In this PR we're just trying to convert the original RESTEasy-based service into a plain servlet so we can drop the RESTEasy dependency later on, but the functionality should be pretty much identical. I think @fmarco76 is trying to fix the hard-coded ACL too, but the behavior should still be unchanged (which is why it's a bit complicated).

If the behavior of the original code itself is not correct we probably should review how this feature is tested and documented.

Reply:
@edewata thanks.  yeah it's okay to create a new follow-up ticket to change the behavior to meet what we need.  Basically "checkOwner" is not needed, as long as the user is a CA agent.  Although it does make one wonder why no one has complained about it.